Fix build time issues with tools who like to export LD_LIBRARY_PATH
to retarded values.

http://bugs.gentoo.org/99593
http://lists.gnu.org/archive/html/libtool/2005-09/msg00053.html

--- monkey/ltmain.in
+++ butt/ltmain.in
@@ -5085,6 +5085,11 @@
 	$echo >> $output "\
     if test \"\$libtool_execute_magic\" != \"$magic\"; then
       # Run the actual program with our arguments.
+
+      # Make sure env LD_LIBRARY_PATH does not mess us up
+      if test -n \"\${LD_LIBRARY_PATH+set}\"; then
+        export LD_LIBRARY_PATH=\$progdir:\$LD_LIBRARY_PATH
+      fi
 "
 	case $host in
 	# Backslashes separate directories on plain windows
